### PR TITLE
auth: add impersonating client

### PR DESCRIPTION
**Pull request by mfojtik**
Used by integration and openshift apiserver (apiserver and admission).

This is the last dependency in tests that point to openshift-apiserver.

/cc @deads2k

### CODE REVIEW

File: pkg/authentication/impersonatingclient/impersonate_rbac.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import (
+	"context"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	rbacv1 "k8s.io/client-go/kubernetes/typed/rbac/v1"
+	"k8s.io/client-go/rest"
+)
+
+func NewImpersonatingRBACFromContext(ctx context.Context, restclient rest.Interface) (rbacv1.RbacV1Interface, error) {
```
Comment (sttts) [inline pkg/authentication/impersonatingclient/impersonate_rbac.go]: godoc

Comment (mfojtik) [reply pkg/authentication/impersonatingclient/impersonate_rbac.go]: fixed

File: pkg/authentication/impersonatingclient/impersonate_rbac.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import (
+	"context"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	rbacv1 "k8s.io/client-go/kubernetes/typed/rbac/v1"
+	"k8s.io/client-go/rest"
+)
+
+// NewImpersonatingRBACFromContext returns an RBAC client based on the current request context.
```
Comment (sttts) [inline pkg/authentication/impersonatingclient/impersonate_rbac.go]: returns an impersonated RBAC client with the user of the context.

**Comment by mfojtik** [comment]
/hold

@deads2k had objections

**Comment by deads2k** [comment]
/close

This is a very advanced power that basically no one should be using.  Only the config creation was used elsewhere, we can copy the five lines.

**Pull request closed without merging**